azure/ConsiderationsForServiceDesign.md

```diff
@@ -141,94 +144,205 @@ cannot collide with a resource path that contains user-specified resource ids.
 an operation may take a significant amount of time -- longer than a client will want to block
 waiting for the result.
-Azure allows for two forms of this design pattern: resource-based long-running operations (RELO),
-which is the preferred pattern, and long-running operations with a status monitor.
 
-In both patterns, the processing of the operation is initiated by one API call and the client
-obtains the results of the operation from a subsequent API call.
-Here we illustrate the sequence of API calls involved in each of these patterns.
+The request that initiates a long-running operation returns a response that points to or embeds
+a _status monitor_, which is an ephemeral resource that will track the status and final result of the operation.
+The status monitor resource is distinct from the target resource (if any) and specific to the individual
+operation request.
+
+A POST or DELETE operation returns a `202 Accepted` response with the status monitor in the response body.
+A long-running POST should not be used for resource create -- use PUT as described below.
+PATCH must never be used for LROs -- it should be reserved for simple resource updates.
+If a long-running update is required it should be implemented with POST.
+
+There is a special form of long-running operation initiated with PUT that is described
+in [Create (PUT) with additional long-running processing](#create-put-with-additional-long-running-processing).
+The remainder of this section describes the pattern for long-running POST and DELETE operations.
+
+This diagram illustrates how a long-running operation with a status monitor is initiated and then how the client
+determines it has completed and obtains its results:
+
+```mermaid
+sequenceDiagram
+    participant Client
+    participant API Endpoint
+    participant Status Monitor
+    Client->>API Endpoint: POST/DELETE
+    API Endpoint->>Client: HTTP/1.1 202 Accepted<br/>Retry-After: 5<br/>{ "id": "22", "status": "NotStarted" }
+    Client->>Status Monitor: GET
+    Status Monitor->>Client: HTTP/1.1 200 Ok<br/>Retry-After: 5<br/>{ "id": "22", "status": "Running" }
+    Client->>Status Monitor: GET
+    Status Monitor->>Client: HTTP/1.1 200 Ok<br/>{ "id": "22", "status": "Succeeded" }
+```
 
-### Resource-based long-running operations
+1. The client sends the request to initiate the long-running operation.
+The initial request could be a PUT, POST, or DELETE method.
+The request may contain an `operation-id` header that the service uses as the ID of the status monitor created for the operation.
+
+2. The service validates the request and initiates the operation processing.
+If there are any problems with the request, the service responds with a `4xx` status code and error response body.
+Otherwise the service responds with a `202-Accepted` HTTP status code.
+The response body is the status monitor for the operation including the ID, either from the request header or generated by the service.
```

Comment:
Should we say that the response "MUST" contains an `id` property (if 202)?

Is `status` also a "MUST" in response of initial request?

Reply:
I am not sure if it is just me, but `id` in response of initial request of POST could be confusing with the same `id` in response of initial request of PUT. The former is the ID for status monitor, the latter is the ID for the resource.

Reply:
POST should never be used to create resources and especially not an LRO POST, so there should be no confusion about the `id` in the response body of LRO POST.

LRO PUT will return the resource in the body and the operation ID in the `Operation-Id` header, so I think this is also clear which ID is for which.

I think this is enough to avoid confusion for most users, but you want to suggest something that would make it even clearer please feel free.

Reply:
I wouldn't say *never*. Consider a Key Vault certificate. The request to create is significantly different from the resource created just by the very nature of CSRs vs. certificates. From meetings, I thought we agree this is rare and maybe doesn't need to be detailed, but shouldn't we at least acknowledge it is a (less desirable) possibility lest we have different teams coming up with different solutions?

Reply:
...in fact, it's documented below and acknowledged by Jeffrey.

Reply:
Appears that I am the only one confused about the `id`. Let's drop it.

---

The POST to create resource would be a different topic.

I assume if we do that, the (created) resource ID (and maybe the full URL of the resource) would be provided in the `result` property.
```
{
   "id": "22",
   "status": "Succeeded",
   "result": { ... }
}
```

Reply:
I added a statement in Guidelines.md:

> :no_entry: **DO NOT** use a long-running POST to create a resource -- use PUT as described below.

If some service insists on using POST rather than PUT we'll probably guide them into a design that looks like the PUT pattern, but we shouldn't offer this as an option in the Guidelines.

Reply:
There are valid uses cases for POST for resource creation. It should be avoided, yes, and certainly not used for service implementation convenience, but for use cases where it is imperative that the service owns the uniqueness of the identity of a resource (possibly global uniqueness), POST would be the preferred pattern. I don't mind this verbiage in the guidelines as long as we, as a stewardship board, understand when it is and when it is not appropriate to use POST for resource creates.